### PR TITLE
caf: events: Add documentation for module state event

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds doxygen documentation for module state event.

Jira: NCSDK-9551

### CODE REVIEW

**Review by emob-nordic — APPROVED** [review]

File: include/caf/events/module_state_event.h [hunk 1]
```diff
@@ -73,9 +95,10 @@ static inline bool module_check_id_valid(size_t module_idx)
 /**
  * @brief Check if there is 0 in all the flags
  *
- * @param mf     Pointer to modules flags variable
- * @retval true  All the flags have value of 0
- * @retval false Any of the flags is not 0
+ * @param[in] mf Pointer to modules flags variable.
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply include/caf/events/module_state_event.h:98]: done

File: include/caf/events/module_state_event.h [hunk 2]
```diff
@@ -90,12 +113,13 @@ static inline bool module_flags_check_zero(const struct module_flags *mf)
 }
 
 /**
- * @brief Test single module bit
+ * @brief Test single module bit.
+ *
+ * @param[in] mf         Pointer to module flags variable.
```
Comment (greg-fer) [inline include/caf/events/module_state_event.h:118]: `module_flags`? Or "Pointer to the variable for the flags module."?

File: include/caf/events/module_state_event.h [hunk 3]
```diff
@@ -90,12 +113,13 @@ static inline bool module_flags_check_zero(const struct module_flags *mf)
...
+ * @param[in] module_idx The index of the module whose bit should be tested.
```
Comment (greg-fer) [inline include/caf/events/module_state_event.h:119]: Avoid `should` in docs of any kind. It's ambiguous.

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: include/caf/events/module_state_event.h (hunk 2 above)
Comment (MarekPieta) [reply include/caf/events/module_state_event.h:118]: This is actually pointer to variable that is used to store application module flags related to something (for example power management restrictions). I think that changing this to your sentence could be a bit misleading.

If you wish, I could change it to "Pointer to variable representing module flags".

File: include/caf/events/module_state_event.h (hunk 3 above)
Comment (MarekPieta) [reply include/caf/events/module_state_event.h:119]: ok, changed

File: include/caf/events/module_state_event.h [hunk 4]
```diff
@@ -127,42 +151,83 @@ static inline void module_flags_set_bit(struct module_flags *mf, size_t module_i
...
+ * Every application module must register for module state event and submit this event when state
+ * of the module changes. An application module can initialize itself when all the required
+ * application modules are initialized (report MODULE_STATE_READY). This ensures proper
+ * initialization order of the application modules.
+ */
 struct module_state_event {
+	/** Event header. */
 	struct event_header header;
 
+	/* ID of the module. */
```
Comment (pdunaj) [inline include/caf/events/module_state_event.h:208]: non doxygen comment

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: include/caf/events/module_state_event.h (hunk 4 above)
Comment (MarekPieta) [reply include/caf/events/module_state_event.h]: Fixed

**Review by pdunaj — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**